### PR TITLE
IGNITE-15754: Fix marshaller exception handling

### PR DESCRIPTION
**Pull request by AMashenkov**
https://issues.apache.org/jira/browse/IGNITE-15754

### CODE REVIEW

File: modules/table/src/main/java/org/apache/ignite/internal/table/RecordBinaryViewImpl.java [hunk 1]
```diff
@@ -265,7 +266,7 @@ public RecordBinaryViewImpl(InternalTable tbl, SchemaRegistry schemaReg, TableMa
     @Override public @NotNull CompletableFuture<Tuple> getAndDeleteAsync(@NotNull Tuple rec) {
         Objects.requireNonNull(rec);
 
-        final Row row = marshaller().marshalKey(rec);
+        final Row row = marshal(rec, true);
```
Comment (ygerzhedovich) [inline modules/table/src/main/java/org/apache/ignite/internal/table/RecordBinaryViewImpl.java:269]:
```suggestion
        final Row keyRow = marshal(rec, true);
```

File: modules/table/src/main/java/org/apache/ignite/internal/table/RecordBinaryViewImpl.java [hunk 2]
```diff
@@ -304,7 +305,7 @@ public RecordBinaryViewImpl(InternalTable tbl, SchemaRegistry schemaReg, TableMa
         HashSet<BinaryRow> keys = new HashSet<>(recs.size());
 
         for (Tuple keyRec : recs) {
-            final Row keyRow = marshaller().marshal(keyRec);
+            final Row keyRow = marshal(keyRec, false);
```
Comment (ygerzhedovich) [inline modules/table/src/main/java/org/apache/ignite/internal/table/RecordBinaryViewImpl.java:308]: Are you sure that should be false? the method name is 'deleteAllExactAsync' so we should check not only key

File: modules/table/src/main/java/org/apache/ignite/internal/table/RecordBinaryViewImpl.java [hunk 3]
```diff
@@ -111,7 +112,7 @@ public RecordBinaryViewImpl(InternalTable tbl, SchemaRegistry schemaReg, TableMa
     @Override public @NotNull CompletableFuture<Void> upsertAsync(@NotNull Tuple rec) {
         Objects.requireNonNull(rec);
 
-        final Row keyRow = marshaller().marshal(rec);
+        final Row keyRow = marshal(rec, false);
```
Comment (ygerzhedovich) [inline modules/table/src/main/java/org/apache/ignite/internal/table/RecordBinaryViewImpl.java:115]:
```suggestion
        final Row row = marshal(rec, false);
```

File: modules/table/src/test/java/org/apache/ignite/internal/table/RecordBinaryViewOperationsTest.java [hunk 4]
```diff
@@ -410,4 +413,17 @@ void assertEqualsValues(SchemaDescriptor schema, Tuple expected, Tuple actual) {
     @NotNull private TableImpl createTableImpl(SchemaDescriptor schema) {
         return new TableImpl(new DummyInternalTableImpl(), new DummySchemaManagerImpl(schema), null);
     }
+
+    public <T extends Throwable> void assertThrowsWithCause(Class<T> expectedType, Executable executable) {
```
Comment (ygerzhedovich) [inline modules/table/src/test/java/org/apache/ignite/internal/table/RecordBinaryViewOperationsTest.java:417]: maybe it's time to introduce some TestUtils. WDYT?

Comment (AMashenkov) [reply modules/table/src/test/java/org/apache/ignite/internal/table/RecordBinaryViewOperationsTest.java:417]: I think we should check a public exception here and error code, once we will introduce codes and public exceptions.
So, an assertThrowsWithCause must be replaced to junit assertThrows with a correct public exception.

We check a cause here, which is an internal exception, as a temporal solution unless we have public classes and error codes.

**Review by ygerzhedovich — APPROVED** [review]

**Pull request merged**